src/oemof/visio/energy_system_graph.py

```diff
@@ -325,30 +417,69 @@ def connect(self, a, b):
         b: `oemof.solph.network.Node`
             An oemof node (usually a Bus or a Component)
         """
-        if not isinstance(a, Bus):
-            a = fixed_width_text(a.label, char_num=self.txt_width)
-        else:
-            a = a.label
-        if not isinstance(b, Bus):
-            b = fixed_width_text(b.label, char_num=self.txt_width)
+        if a.depth <= self.max_depth and b.depth <= self.max_depth:
+            if not isinstance(a, Bus):
+                a = fixed_width_text(str(a.label), char_num=self.txt_width)
+            else:
+                a = str(a.label)
+            if not isinstance(b, Bus):
+                b = fixed_width_text(str(b.label), char_num=self.txt_width)
+            else:
+                b = str(b.label)
+
+            self.dot.edge(a, b)
+
+    def _generate_graph(self, max_depth=None):
+        self.dot = graphviz.Digraph(format=self.img_format, **self.digraph_kwargs)
+        self.busses = []
+        self.max_depth_connexions = []
+        if max_depth is not None:
+            self.max_depth = max_depth
         else:
-            b = b.label
+            self.max_depth = max([n.depth for n in self.energy_system.nodes])
+
+        if self.legend is True:
+            with self.dot.subgraph(name="cluster_1") as c:
+                # color of the legend box
+                c.attr(color="black")
+                # title of the legend box
+                c.attr(label="Legends")
+                self.add_bus(subgraph=c)
+                self.add_sink(subgraph=c)
+                self.add_source(subgraph=c)
+                self.add_transformer(subgraph=c)
+                self.add_storage(subgraph=c)
 
-        self.dot.edge(a, b)
+        # draw a node for each of the energy_system's component.
+        # the shape depends on the component's type.
+        self.add_components(self.energy_system.nodes)
 
-    def view(self, **kwargs):
+        for link in self.max_depth_connexions:
+            self.connect(*link)
+
+    def view(self, max_depth=None, **kwargs):
         """Call the view method of the DiGraph instance"""
+        self._generate_graph(max_depth)
         self.dot.view(**kwargs)
 
-    def render(self, **kwargs):
+    def render(self, max_depth=None, **kwargs):
         """Call the render method of the DiGraph instance"""
-        print(self.dot.render(**kwargs))
+        self._generate_graph(max_depth)
+        fname = self.dot.render(cleanup=True, **kwargs)
+        logging.info(
+            f"The energy system graph was saved under '{fname}' in the current directory"
+        )
         return self.dot
 
-    def pipe(self, **kwargs):
+    def pipe(self, max_depth=None, **kwargs):
         """Call the pipe method of the DiGraph instance"""
+        self._generate_graph(max_depth)
         self.dot.pipe(**kwargs)
 
+    def source(self, max_depth=None):
```

Comment:
In order to use this to run tests one would need to test with different version of oemof-network I believe